gateway-service-impl/src/main/java/org/hypertrace/gateway/service/explore/TimeAggregationsWithGroupByRequestHandler.java

```diff
@@ -228,7 +230,14 @@ private void buildInClauseFilterValue(Value value, Value.Builder valueBuilder) {
         valueBuilder.addStringArray(value.getString());
         break;
       case STRING_ARRAY:
-        valueBuilder.addAllStringArray(value.getStringArrayList());
+        // Handle special case when "null" string is returned as string array value(default value
+        // scenario). This will be converted to empty list in value converter. In such a case use
+        // list with "null" value
+        if (value.getStringArrayList().isEmpty()) {
```

Comment:
> > so one behavior discrepancy this would lead to to - let's say we have 10 rows, and 9 of them have no values for a multivalued col X and 1 of them does. Both before and after this change, if we grouped by X we'd get a single returned grouped with a count of 1.
> 
> From the Query Service (QS) point of view, it will return two groups. Let's assume that 1 row which is not empty has value - ["hello"]. The output will be as below. Returned groups:
> 
> ```
> {
>   {"null", 9},
>   {"hello", 1}
> }
> ```
> 
> And if the none empty row of X has more than one value, say - ["hello", "world"], QS will return 3 groups:
> 
> ```
> {
>   {"null", 9},
>   {"hello", 1},
>   {"world", 1}
> }
> ```
> 
> Moving on to Pinot, if the column (or multi-value column) has no value, it will return the default value. The default value for a String column is `null`. For a multi-value column, if not preset, it uses the default of a single-value column of the same type.

Reply:
The described behavior makes sense, the first issue to resolve however is how GW should behave because it's not consistent. If we do a group by with time series, it won't return that null (or empty list); if we do it without, it returns an empty list as one value.

Reply:
Agreed that both should return empty list